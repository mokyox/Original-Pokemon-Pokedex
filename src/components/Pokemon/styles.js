import styled from "styled-components";
import { device } from "../../styles/media";

export const Container = styled.div`
    margin: 0 auto;
    padding: 4rem;
    font-size: 0.75rem;
    @media ${device.tablet} {
        font-size: 1.1rem;
    }
`;
