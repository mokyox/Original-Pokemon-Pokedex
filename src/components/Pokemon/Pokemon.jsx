import React from "react";
import { Container } from "./styles";
import { PokemonTypeBadge } from "../PokemonCard/styles";

const Pokemon = ({ name, id, image, type, stats, description, ability }) => {
    return (
        <Container>
            <div className="card bg-dark pt-2 p-3 mx-auto">
                <div className="text-nowrap p-1">
                    <h1>{name}</h1>
                    <h3>#{id}</h3>
                    {type.map((type) => (
                        <PokemonTypeBadge
                            className="p-1 m-1 badge"
                            key={type}
                            type={type}
                        >
                            {type}
                        </PokemonTypeBadge>
                    ))}
                </div>
                <div className="card-body">
                    <img
                        src={image}
                        alt={`pokemon: ${name}`}
                        height="96px"
                        width="96px"
                    />
                    {Object.entries(stats).map(([stat, value]) => {
                        return (
                            <div key={stat} className="p-2">
                                <span className="font-weight-bold">{stat}</span>
                                <span>:{value}</span>
                            </div>
                        );
                    })}

                    <span className="text-capitalize p-3">
                        <span className="font-weight-bold">Abilities: </span>
                        {ability}
                    </span>
                    <p className="p-1">{description}</p>
                </div>
            </div>
        </Container>
    );
};

export default Pokemon;
