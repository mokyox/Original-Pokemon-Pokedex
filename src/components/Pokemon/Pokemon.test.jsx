import React from "react";
import Pokemon from "./Pokemon";
import renderer from "react-test-renderer";

describe("<Pokemon/>", () => {
  const pokemonInfo = {
    id: 1,
    name: "Bulbasaur",
    image: "",
    type: ["Grass", "Poison"],
    description:
      "Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sunâ€™s rays, the seed grows progressively larger.",
    ability: ["Chlorophyll", "Overgrow"],
    stats: {
      HP: 45,
      Attack: 49,
      Defense: 49,
      "Sp. Attack": 65,
      "Sp. Defense": 65,
      Speed: 45,
    },
  };
  it("renders correctly", () => {
    const tree = renderer.create(<Pokemon {...pokemonInfo} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
