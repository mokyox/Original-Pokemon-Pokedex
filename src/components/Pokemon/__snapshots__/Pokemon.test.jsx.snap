// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pokemon/> renders correctly 1`] = `
<div
  className="sc-gsTCUz bdNDAl"
>
  <div
    className="sc-bdfBwQ kijUgU card bg-dark"
  >
    <div
      className="text-nowrap"
    >
      <h1>
        Bulbasaur
      </h1>
      <h3>
        #
        1
      </h3>
      <span
        className="sc-gKsewC iVbtrA p-1 m-1 badge"
        type="Grass"
      >
        Grass
      </span>
      <span
        className="sc-gKsewC cJxrDx p-1 m-1 badge"
        type="Poison"
      >
        Poison
      </span>
    </div>
    <div
      className="card-body"
    >
      <img
        alt="pokemon: Bulbasaur"
        height="96px"
        src=""
        width="96px"
      />
      <p
        className="text-capitalize"
      >
        <span
          className="font-weight-bold"
        >
          Abilities: 
        </span>
        Chlorophyll
        Overgrow
      </p>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          HP
        </span>
        <span>
          : 
          45
        </span>
      </div>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          Attack
        </span>
        <span>
          : 
          49
        </span>
      </div>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          Defense
        </span>
        <span>
          : 
          49
        </span>
      </div>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          Sp. Attack
        </span>
        <span>
          : 
          65
        </span>
      </div>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          Sp. Defense
        </span>
        <span>
          : 
          65
        </span>
      </div>
      <div
        className="p-2"
      >
        <span
          className="font-weight-bold"
        >
          Speed
        </span>
        <span>
          : 
          45
        </span>
      </div>
      <p
        className="p-1 mt-2"
      >
        Bulbasaur can be seen napping in bright sunlight. There is a seed on its back. By soaking up the sunâ€™s rays, the seed grows progressively larger.
      </p>
    </div>
  </div>
  <div
    className="sc-dlfnbm jDSPV"
  >
    
    <a
      className="text-monospace text-uppercase p-2 m-2 badge badge-dark"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Go home
    </a>
    <a
      className="text-monospace text-uppercase p-2 m-2 badge badge-dark"
      href="/pokemon/2"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      Next Pokemon
    </a>
  </div>
</div>
`;
